plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -292,10 +297,21 @@ func (s *serviceAccount) limitSecretReferences(serviceAccount *api.ServiceAccoun
 			continue
 		}
 		secretName := source.Secret.SecretName
-		if !referencedSecrets.Has(secretName) {
+		if !mountableSecrets.Has(secretName) {
```

Comment:
> Is there a use case where you have a service account which has pull secrets, but you want to use a specific pod with a pull secret which doesn't belong to the service account?

That's a more general question about secret protection.  Our current statement is that if you're going to use a secret and you have service accounts enabled, the service account should list that secret as allowed.

Loosening that statement to consider the union of the requestor *and* the service account make some bit easier, but makes it more dangerous to allow any privileged agent to create pods.  I think I'd like to keep the current statement for the near term and perhaps reconsider once we have a more complete impersonation story.